source/server/http_test_server_filter.h

```diff
@@ -73,6 +73,7 @@ class HttpTestServerDecoderFilter : public Envoy::Http::StreamDecoderFilter {
   Envoy::Http::StreamDecoderFilterCallbacks* decoder_callbacks_;
   nighthawk::server::ResponseOptions base_config_;
   absl::optional<std::string> error_message_;
+  absl::optional<std::string> request_headers_dump_;
```

Comment:
I am not familiar with this portion of the code - can you confirm that the lifetime of the HttpTestServerDecoderFilter class is per single request?